dangerzone/logic.py

```diff
@@ -64,16 +64,20 @@ def clear_documents(self) -> None:
     def convert_documents(
         self, ocr_lang: Optional[str], stdout_callback: Optional[Callable] = None
     ) -> None:
-        def convert_doc(document: Document) -> None:
-            self.isolation_provider.convert(
-                document,
-                ocr_lang,
-                stdout_callback,
-            )
-
-        max_jobs = self.isolation_provider.get_max_parallel_conversions()
-        with concurrent.futures.ThreadPoolExecutor(max_workers=max_jobs) as executor:
-            executor.map(convert_doc, self.documents)
+        try:
+            def convert_doc(document: Document) -> None:
+                self.isolation_provider.convert(
+                    document,
+                    ocr_lang,
+                    stdout_callback,
+                )
+                        
+            max_jobs = self.isolation_provider.get_max_parallel_conversions()
+            with concurrent.futures.ThreadPoolExecutor(max_workers=max_jobs) as executor:
+                executor.map(convert_doc, self.documents)
+        except Exception as e:
+            print(e)
+            log.exception(f"Error occured while converting doc")
```

Comment:
... but not that high up. The original issue is that `executor.map()` swallows exceptions by design. This means that we don't gain something by wrapping `executor.map()` in a try...except block. Instead, we need to catch exceptions within the `convert_doc()` function, which is the highest an exception from the conversion may reach.

Something like :

```suggestion
      def convert_doc(document: Document) -> None:
          try
                self.isolation_provider.convert(
                    document,
                    ocr_lang,
                    stdout_callback,
                )
            except Exception:
                log.exception(f"Unexpected error occurred while converting '{document}'")
                document.mark_as_failed()
```

Reply:
Thanks I updated the code so it is contained within the convert_doc function and the unit tests are passing